pandas/tests/reshape/merge/test_multi.py

```diff
@@ -0,0 +1,623 @@
+# pylint: disable=E1103
+
+from collections import OrderedDict
+
+import numpy as np
+import pytest
+from numpy import nan
+from numpy.random import randn
+
+import pandas as pd
+import pandas.util.testing as tm
+from pandas import (DataFrame, Index, MultiIndex, Series)
+from pandas.compat import lzip
+from pandas.core.reshape.concat import concat
+from pandas.core.reshape.merge import merge
+
+
+@pytest.fixture
+def left():
+    # a little relevant example with NAs
+    key1 = ['bar', 'bar', 'bar', 'foo', 'foo', 'baz', 'baz', 'qux',
+            'qux', 'snap']
+    key2 = ['two', 'one', 'three', 'one', 'two', 'one', 'two', 'two',
+            'three', 'one']
+
+    data = np.random.randn(len(key1))
+    return DataFrame({'key1': key1, 'key2': key2, 'data': data})
+
+
+@pytest.fixture
+def right():
+    index = MultiIndex(levels=[['foo', 'bar', 'baz', 'qux'],
+                               ['one', 'two', 'three']],
+                       labels=[[0, 0, 0, 1, 1, 2, 2, 3, 3, 3],
+                               [0, 1, 2, 0, 1, 1, 2, 0, 1, 2]],
+                       names=['first', 'second'])
+
+    return DataFrame(np.random.randn(10, 3), index=index,
+                     columns=['j_one', 'j_two', 'j_three'])
+
+
+class TestMergeMulti(object):
+
+    def test_merge_on_multikey(self, left, right):
+        joined = left.join(right, on=['key1', 'key2'])
+
+        join_key = Index(lzip(left['key1'], left['key2']))
+        indexer = right.index.get_indexer(join_key)
+        ex_values = right.values.take(indexer, axis=0)
+        ex_values[indexer == -1] = np.nan
+        expected = left.join(DataFrame(ex_values, columns=right.columns))
+
+        # TODO: columns aren't in the same order yet
+        tm.assert_frame_equal(joined, expected.loc[:, joined.columns])
+
+        left = left.join(right, on=['key1', 'key2'], sort=True)
+        right = expected.loc[:, joined.columns].sort_values(['key1', 'key2'],
+                                                            kind='mergesort')
+        tm.assert_frame_equal(left, right)
+
+    def test_left_join_multi_index(self):
```

Comment:
Generally I find this test function hard to decipher - can you provide some commentary around what the comparisons of `4th` and `5th` are supposed to yield? It may also make sense to split off the NA testing for more logical test separation

Reply:
I did not write these set of tests and I didn't manage to decipher them either.
Shall I remove (or comment out) until we get some clarifications from the original author?